File: app/MyJobPitch-New/app/src/main/java/com/myjobpitch/api/data/Message.java
package com.myjobpitch.api.data;

import com.myjobpitch.api.MJPAPIObject;

import java.util.Date;

/**
 * Created by Jamie on 27/05/2015.
 */
public class Message extends MJPAPIObject {
    protected Integer application;
    protected Boolean system;
    protected Integer from_role;
    protected String content;
    protected Date created;
    protected Boolean read;

    public Integer getApplication() {
        return application;
    }

    public Boolean getSystem() {
        return system;
    }

    public Integer getFrom_role() {
        return from_role;
    }

    public String getContent() {
        return content;
    }

    public Date getCreated() {
        return created;
    }

    public Boolean getRead() {
        return read;
    }
}

File: ios/MyJobPitch/Utils/API/models/Message.swift
//
//  Message.swift
//  MyJobPitch
//
//  Created by dev on 12/20/16.
//  Copyright © 2016 myjobpitch. All rights reserved.
//

import Foundation

class Message: MJPObject {
    
    static let mappingArray = [ "id", "system", "content", "read", "created", "application", "interview" ]
    static let mappingDictionary = [ "fromRole": "from_role" ]

    var system = false
    var content: String!
    var read = false
    var created: Date!
    var application: NSNumber!
    var fromRole: NSNumber!
    
    // API V3
    var interview: NSNumber!
}

class MessageForCreation: NSObject {
    
    static let mappingArray = [ "application", "content" ]
    
    var content: String!
    var application: NSNumber!
}

class MessageForUpdate: MJPObject {
    
    static let mappingArray = [ "read" ]

    var read = true
}
